core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -2043,6 +2043,16 @@ public FormValidation doCheckAssignedLabelString(@AncestorInPath AbstractProject
 
         public FormValidation doCheckLabel(@AncestorInPath AbstractProject<?,?> project,
                                            @QueryParameter String value) {
+            return validateLabelExpression(value, project);
+        }
+
+        /**
+         * Validate label expression string.
+         *
+         * @param project May be specified to perform project specific validation.
+         * @since TODO
+         */
+        public static @Nonnull FormValidation validateLabelExpression(String value, @CheckForNull AbstractProject<?, ?> project) {
```

Comment:
I wish I had noticed this PR when it was filed, because it should really be taking `Job` not `AbstractProject`.